lxd/networks.go

```diff
@@ -2151,3 +2152,74 @@ func (n *network) spawnForkDNS(listenAddress string) error {
 
 	return nil
 }
+
+// refreshForkdnsServerAddresses retrieves the IPv4 address of each cluster node (excluding ourselves)
+// for this network. It then updates the forkdns server list file if there are changes.
+func (n *network) refreshForkdnsServerAddresses() {
+	addresses := []string{}
+
+	// Get the list of nodes
+	nodes, err := cluster.List(n.state)
+	if err != nil {
+		logger.Errorf("Failed to get forkdns cluster list for '%s': %v", n.name, err)
+		return
+	}
+
+	localAddress, err := node.HTTPSAddress(n.state.Node)
+	if err != nil {
+		logger.Errorf("Failed to get forkdns local cluster address for '%s': %v", n.name, err)
+		return
+	}
+
+	cert := n.state.Endpoints.NetworkCert()
+	for _, node := range nodes {
+		address := strings.TrimPrefix(node.URL, "https://")
+		if address == localAddress {
+			continue
+		}
+
+	again:
```

Comment:
Why the fancy retry code when the task itself is being run so regularly? Isn't it easier to just log an error and return, then let LXD trigger the next round?

My understanding of our task code is that it's triggering the task at the interval defined regardless of when it's completed and whether it's still running, so having built-in retry may actually cause the task to be run multiple times.

Reply:
@stgraber I had left it in from the original implementation that had retries at start up. But yes, there is no need for this anymore. I will remove.

Reply:
@stgraber I have updated to just log an error and return if we cannot connect to the remote node. I was thinking about just logging an error and then moving onto the next node (which would result in the node we couldnt connect to being removed from the forkdns servers list). 

However I was thinking this might just be a temporary API issue and if we remove it from forkdns servers list too quickly this could disrupt DNS queries. If it is down for a long time will it get removed from the cluster node candidates list entirely? ```nodes, err := cluster.List(n.state)```?